staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -594,6 +599,35 @@ func (w *watchCache) GetAllEventsSinceThreadUnsafe(resourceVersion uint64) ([]*w
 		// current state and only then start watching from that point.
 		//
 		// TODO: In v2 api, we should stop returning the current state - #13969.
+		ci := &watchCacheInterval{fillAll: true}
```

Comment:
Regarding the fillAll argument that I added below.

The RV=0 is relatively rare case (IIRC only kubectl uses thatin some path). So I wouldn't bother that much about it now.

As I commented below - we will need a buffer inside interval anyway.
So given we need that - let's just simply list here, put all the items into that buffer and return it.


This doesn't address the problem for RV=0, but it's not a common problem and I would prefer more incremental approach to reduce the risk (and RV=0 doesn't seem to be a big problem).